regex_test/regexpatterns_test.go

```diff
@@ -0,0 +1,223 @@
+	for n := 0; n < b.N; n++ {
+		reMatch(regexToMatch.Copy(), lineNoMatch, numGoroutines)
+	}
+}
+
+func Benchmark_CyclePattern_Simple_Match(b *testing.B) {
+	rePool := make([]*regexp.Regexp, poolSize)
+	for i := 0; i < poolSize; i++ {
+		regexToMatch, err := regexp.Compile(simpleRegexp)
+		if err != nil {
+			b.Fatal("failed to compile regex")
+		}
+
+		rePool[i] = regexToMatch
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		reCycleMatch(rePool, lineMatch, numGoroutines)
+	}
+}
+
+func Benchmark_CyclePattern_Simple_NoMatch(b *testing.B) {
+	rePool := make([]*regexp.Regexp, poolSize)
+	for i := 0; i < poolSize; i++ {
+		regexToMatch, err := regexp.Compile(simpleRegexp)
+		if err != nil {
+			b.Fatal("failed to compile regex")
+		}
+
+		rePool[i] = regexToMatch
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		reCycleMatch(rePool, lineNoMatch, numGoroutines)
+	}
+}
+
+func Benchmark_OriginalPattern_Complex_Match(b *testing.B) {
+	regexToMatch, err := regexp.Compile(complexRegexp)
+	if err != nil {
+		b.Fatal("failed to compile regex")
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		reMatch(regexToMatch, lineMatch, numGoroutines)
+	}
+}
+
+func Benchmark_OriginalPattern_Complex_NoMatch(b *testing.B) {
+	regexToMatch, err := regexp.Compile(complexRegexp)
+	if err != nil {
+		b.Fatal("failed to compile regex")
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		reMatch(regexToMatch, lineNoMatch, numGoroutines)
+	}
+}
+
+
+func Benchmark_CopyPattern_Complex_Match(b *testing.B) {
+	regexToMatch, err := regexp.Compile(complexRegexp)
+	if err != nil {
+		b.Fatal("failed to compile regex")
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		reMatch(regexToMatch.Copy(), lineMatch, numGoroutines)
+	}
+}
+
+func Benchmark_CopyPattern_Complex_NoMatch(b *testing.B) {
+	regexToMatch, err := regexp.Compile(complexRegexp)
+	if err != nil {
+		b.Fatal("failed to compile regex")
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		reMatch(regexToMatch.Copy(), lineNoMatch, numGoroutines)
+	}
+}
+
+func Benchmark_CyclePattern_Complex_Match(b *testing.B) {
+	rePool := make([]*regexp.Regexp, poolSize)
+	for i := 0; i < poolSize; i++ {
+		regexToMatch, err := regexp.Compile(complexRegexp)
+		if err != nil {
+			b.Fatal("failed to compile regex")
+		}
+
+		rePool[i] = regexToMatch
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		reCycleMatch(rePool, lineMatch, numGoroutines)
+	}
+}
+
+func Benchmark_CyclePattern_Complex_NoMatch(b *testing.B) {
+	rePool := make([]*regexp.Regexp, poolSize)
+	for i := 0; i < poolSize; i++ {
+		regexToMatch, err := regexp.Compile(complexRegexp)
+		if err != nil {
+			b.Fatal("failed to compile regex")
+		}
+
+		rePool[i] = regexToMatch
+	}
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		reCycleMatch(rePool, lineNoMatch, numGoroutines)
+	}
+}
+
+
+// Results
```

Comment:
OK interesting - these results indicate that there's negligible difference between the approaches. Which would mean that the implementation doesn't introduce an unnecessary bottleneck on filtering. Good to know!

I'll re-run on my machine to confirm too.